typespec-tests/src/main/java/com/cadl/armresourceprovider/models/ChildExtensionResourceInterfaces.java

```diff
@@ -40,37 +40,6 @@ Response<ChildExtensionResource> getWithResponse(String resourceUri, String topL
      */
     ChildExtensionResource get(String resourceUri, String topLevelArmResourceName, String childExtensionResourceName);
 
-    /**
-     * Update a ChildExtensionResource.
-     * 
-     * @param resourceUri The fully qualified Azure Resource manager identifier of the resource.
-     * @param topLevelArmResourceName arm resource name for path.
-     * @param childExtensionResourceName ChildExtensionResources.
-     * @param properties The resource properties to be updated.
-     * @param context The context to associate with this operation.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws com.azure.core.management.exception.ManagementException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return extensionResource of Top Level Arm Resource along with {@link Response}.
-     */
-    Response<ChildExtensionResource> updateWithResponse(String resourceUri, String topLevelArmResourceName,
-        String childExtensionResourceName, Object properties, Context context);
-
-    /**
-     * Update a ChildExtensionResource.
-     * 
-     * @param resourceUri The fully qualified Azure Resource manager identifier of the resource.
-     * @param topLevelArmResourceName arm resource name for path.
-     * @param childExtensionResourceName ChildExtensionResources.
-     * @param properties The resource properties to be updated.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws com.azure.core.management.exception.ManagementException thrown if the request is rejected by server.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return extensionResource of Top Level Arm Resource.
-     */
-    ChildExtensionResource update(String resourceUri, String topLevelArmResourceName, String childExtensionResourceName,
-        Object properties);
-
     /**
```

Comment:
This is expected. As we now have `bodyClientModel` (instead of Object):
https://github.com/Azure/autorest.java/blob/9b59bd39524bd50b79d04458b464d5d4a241680b/fluentgen/src/main/java/com/azure/autorest/fluent/mapper/ResourceParser.java#L235-L247

Thus, the update methods should be in resource model's API, not resource collection's.

Reply:
Turns out the cause be a little different. 
Since the found method is actually `PUT`, we do have `resourceUpdate`. The `update` methods in the resource collection are not considered resource update methods, hence not be removed. 

The new generated code should be right.